services/QuillLMS/app/models/classroom_unit_activity_state.rb

```diff
@@ -75,18 +75,18 @@ def update_lessons_cache_with_data
   end
 
   private def handle_pinning
-    if pinned == true
-      if visible == false
-        # unpin ca before archiving
-        update!(pinned: false)
-      else
-        # unpin any other pinned ca before pinning new one
-        classroom = classroom_unit.classroom
-        classroom_unit_ids = classroom.classroom_units.ids.flatten
-        pinned_cua = ClassroomUnitActivityState.unscoped.find_by(pinned: true, classroom_unit_id: classroom_unit_ids)
-        return if pinned_cua && pinned_cua == self
-        pinned_cua.update_column("pinned", false) if pinned_cua
-      end
+    return unless pinned == true
```

Comment:
If this doesn't need to avoid other truthy values (i doubt this is purposeful), this could be:
```suggestion
    return unless pinned
```